modules/cnf-image-based-upgrade-troubleshooting.adoc

```diff
@@ -0,0 +1,289 @@
+// * edge_computing/image-based-upgrade/ztp-image-based-upgrade.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="ztp-image-based-upgrade-troubleshooting_{context}"]
+= Troubleshooting image-based upgrades with {lcao}
+
+You can encounter issues during the image-based upgrade.
+
+[id="ztp-image-based-upgrade-troubleshooting-must-gather_{context}"]
+== Collecting logs
+
+You can use the `oc adm must-gather` CLI to collect information for debugging and troubleshooting.
+
+.Procedure
+
+. Collect data about the Operators by running the following command:
++
+[source,terminal]
+----
+$  oc adm must-gather \
+--dest-dir=must-gather/tmp \
+--image=$(oc -n openshift-lifecycle-agent get deployment.apps/lifecycle-agent-controller-manager -o jsonpath='{.spec.template.spec.containers[?(@.name == "manager")].image}') \
+--image=quay.io/konveyor/oadp-must-gather:latest \ <1>
+--image=quay.io/openshift/origin-must-gather:latest <1>
+----
+<1> (Optional) You can add these options if you need to gather more information from the OADP and the SR-IOV Operators.
+
+[id="ztp-image-based-upgrade-troubleshooting-manual-cleanup_{context}"]
+== AbortFailed or FinalizeFailed error
+
+Issue::
+During the finalize stage or when you stop the process at the `Prep` stage, normally, {lcao} cleans up the following resources:
+
+* Stateroot that is no longer required
+* Precaching resources
+* OADP CRs
+* `ImageBasedUpgrade` CR
+
+If the {lcao} fails to perform the above steps, it transitions to the `AbortFailed` or `FinalizeFailed` states.
+The condition message and log show which steps failed.
+
+.Example error message
+[source,yaml]
+----
+message: failed to delete all the backup CRs. Perform cleanup manually then add 'lca.openshift.io/manual-cleanup-done' annotation to ibu CR to transition back to Idle
+      observedGeneration: 5
+      reason: AbortFailed
+      status: "False"
+      type: Idle
+----
+
+Resolution::
+
+. Inspect the logs to determine why the failure occurred.
+
+. To prompt {lcao} to retry the cleanup, add the `lca.openshift.io/manual-cleanup-done` annotation to the `ImageBasedUpgrade` CR.
+
+After observing this annotation, {lcao} retries the cleanup and if it is successful, the `ImageBasedUpgrade` stage transitions to `Idle`.
+
+If the cleanup fails again, you can manually clean up the resources.
+
+[id="ztp-image-based-upgrade-troubleshooting-stateroot_{context}"]
+=== Cleaning up stateroot manually
+
+Issue::
+
+Stopping at the `Prep` stage, {lcao} cleans up the new stateroot, and finalizing after a successful upgrade or a rollback, normally, {lcao} cleans up the old stateroot.
+If this step fails, it is recommended that you inspect the logs to determine why the failure occurred. 
+
+Resolution::
+
+. Check if there are any existing deployments in the stateroot by running the following command:
++
+[source,terminal]
+----
+$ ostree admin status
+----
+
+. If there are any, clean up the existing deployment:
```

Comment:
Not sure the word "any" is appropriate here.

Below is what I get on idle ibu:
`sh-5.1# ostree admin status                                                                                                                      
* rhcos 427c05e15a0c26d11a864d36290693466f6ab9c9b92305a42e0d3d23a233ecbc.0
    origin: <unknown origin type>          
sh-5.1# `

Reply:
Changed to as per [upstream docs](https://github.com/openshift-kni/lifecycle-agent/blob/main/docs/troubleshooting.md#stateroot-cleanup):
> Clean up the existing deployment: